ATParser.cpp

```diff
@@ -155,7 +155,8 @@ int ATParser::vscanf(const char *format, va_list args)
         sscanf(_buffer+offset, _buffer, &count);
 
         // We only succeed if all characters in the response are matched
-        if (count == j) {
+        // and we are at the end of the response
+        if ((count == j) && (format[i-1] == (const char)c)) {
```

Comment:
Ciao @geky, unfortunately your assumption (`count == j`  **=>** `format[i-1] == (const char)c`) is not correct in an use case of mine where the latest element of an `AT` command response is e.g. an integer value to be read by `%d`. In this case I needed to add another character to the expected response string which because of its syntax could only be a whitespace character and therefore  and due to the specs of `sscanf()` I needed to add the `format[i-1] == (const char)c` check to recognize the end of the integer value.
For exactly the same reason as argued by yourself I believe/hope that this modification is backward compatible.